### ISSUE TITLE
Raspi-4 vs. Raspi-3 issue

### ISSUE THREAD

**Issue by danielcloran** [hzeller/rpi-rgb-led-matrix#1065]
Hzeller
 
Thank you for the incredible library, I have used it to run a box of 5 64x64 led panels for about a year now with no issues until now. 

I recently bought a raspberry pi 4 to hopefully allow for less flickering while my background processes are running on the pi and the lighting has gone weird. See attached photo
<img width="470" alt="Screen Shot 2020-05-10 at 9 59 31 PM" src="https://user-images.githubusercontent.com/29779316/81517632-88f57d80-9309-11ea-95e2-2bf482df416c.png">
. 

It seems as if every two lines is lit up.

I am running debian jessie lite on the raspberry pi 3b+ and running debian buster lite on the raspberry pi 4... I followed the exact same install procedure and cloned straight from the current branch. 

My runtime command is: sudo ./demo -D11 --led-rows=64 --led-cols=64 --led-chain=5 --led-panel-type=FM6126A

I would really appreciate any help you could provide as I've spent the last couple of days banging my head against the wall.

Many thanks,
Daniel Cloran

**Comment by hzeller** [comment]
So you say it works fine in the Pi3 but not the Pi4 ?
The Pi4 is faster, so `--led-slowdown-gpio` might need to be higher.

How are you connecting the panels, are you using an [active3 adapter](https://github.com/hzeller/rpi-rgb-led-matrix/tree/master/adapter/active-3) ? If so, I would also suggest to use parallel chains, e.g use `--led-parallel=3 --led-chain=2` which allows for higher refresh.

**Comment by danielcloran** [comment]
Leds are connected using the red board you recommended in wiring... works perfectly normal and fine on Pi3. I used --led-slowdown-gpio=2 through 4 which helped with the flicker but did not change the lines. 

I have tried using --led-parallel=3 and --led-chain=2 but I got many issues with only half of the panels lighting up so I postponed getting that to work to a later date.

**Comment by hzeller** [comment]
this is strange; did you disable all potential services that might mess with your GPIO pins (such as one wire etc. mentioned in the troubleshooting section).

I've heard a few success reports of people happier with the DietPi distribution as it is more bare-bone and contains less things that might interfere with GPIO pins.

Only every other line or double line with double space showing up is usually a problem with one of the ABCDE lines not properly connected, but if you use the same adapter board and same LED matrix and stick it on different Pis, with only one having a problem, then at least the variable (that the adapter board might be broken) is out.

Are all your grounds between LED panel and Pi properly connected ? Usually that should be taken care of the adapter board, but it can't harm to connect the ground of your supply of the Matrix with the ground of the Pi (if you are using different supplies).

**Comment by danielcloran** [comment]
So I downloaded DietPi last night with a clean install as I saw you mention it elsewhere and a clean build and once again had the same two line issue. Which to me ruled out any Debian buster update that might be hurting it. 

I disabled everything mentioned in troubleshooting (this pi did not have one wire enabled)... and earlier today I retested the setup on the Pi3 and everything still worked fine.

Grounds are connected through the adapter board but I just ran a manual jumper between the two grounds and nothing changed :/ 

Notable to mention is that this is a completely clean install of Debian Buster lite without anything on the pi other than your library and the essentials needed to get and build it.

**Comment by danielcloran** [comment]
I could possibly order another Pi4 just to see if I got some weird defect...

**Comment by hzeller** [comment]
it does sound like some GPIO issue, yep. I once destroyed part of my GPIO pins by accidentally putting on an adapter board shifted by one pin ...

**Comment by danielcloran** [comment]
Argh okay... I'll put an order in now. I'll update this when I test it with the fresh Pi.

Thanks for the help,
Daniel Cloran

**Comment by mikecann** [comment]
@danielcloran I too am building a cube :) we should have a chat sometime and swap notes. As for your particular issue you may want to see this thread: https://github.com/hzeller/rpi-rgb-led-matrix/issues/989

**Comment by danielcloran** [comment]
New raspberry pi 4 arrived today and fixed the problem! I assume I blew one of the GPIO pins on the other one somehow... will be more careful in the future!
Thanks for all the help @hzeller.

@mikecann I'd love to chat! I looked through your thread when trying to run the panels in parallel... but didn't get very far as my panels had some different issues. I'll email you when I get a chance tonight.

**Comment by mikecann** [comment]
@danielcloran ye no worries mate, my email is: mike.cann@gmail.com hit me up :)

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
